tests/components/upnp/test_config_flow.py

```diff
@@ -52,11 +60,39 @@ async def test_flow_ssdp_discovery(hass: HomeAssistantType):
         assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
         assert result["title"] == mock_device.name
         assert result["data"] == {
-            "st": mock_device.device_type,
-            "udn": mock_device.udn,
+            CONFIG_ENTRY_ST: mock_device.device_type,
+            CONFIG_ENTRY_UDN: mock_device.udn,
         }
 
 
+async def test_flow_ssdp_discovery_incomplete(hass: HomeAssistantType):
+    """Test config flow: incomplete discovery through ssdp."""
+    udn = "uuid:device_1"
+    mock_device = MockDevice(udn)
+    discovery_infos = [
+        {
+            DISCOVERY_ST: mock_device.device_type,
+            DISCOVERY_UDN: mock_device.udn,
+            DISCOVERY_LOCATION: "dummy",
+        }
+    ]
+    with patch.object(
+        Device, "async_create_device", AsyncMock(return_value=mock_device)
+    ), patch.object(Device, "async_discover", AsyncMock(return_value=discovery_infos)):
+        # Discovered via step ssdp.
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_SSDP},
+            data={
+                ssdp.ATTR_SSDP_ST: mock_device.device_type,
+                # ssdp.ATTR_UPNP_UDN: mock_device.udn,  # Not provided.
+                "friendlyName": mock_device.name,
+            },
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "incomplete_discovery"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)